src/test/java/uk/gov/justice/hmpps/probationteams/repository/LocalDeliveryUnit2RepositoryTest.kt

```diff
@@ -0,0 +1,184 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.security.test.context.support.WithAnonymousUser
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.context.transaction.TestTransaction
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.hmpps.probationteams.model.LocalDeliveryUnit2
+import uk.gov.justice.hmpps.probationteams.model.ProbationTeam
+import uk.gov.justice.hmpps.probationteams.utils.uniqueLduCode
+import java.util.*
+
+@ActiveProfiles("test")
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@Transactional
+@WithAnonymousUser
+
+class LocalDeliveryUnit2RepositoryTest(
+        @Autowired val repository: LocalDeliveryUnit2Repository,
+        @Autowired val jdbcTemplate: JdbcTemplate
+) {
+
+    @Test
+    fun `Persist Local Delivery Unit`() {
+        val lduCode = uniqueLduCode()
+        val ldu = LocalDeliveryUnit2(
+                probationAreaCode = "ABC",
+                localDeliveryUnitCode = lduCode,
+                functionalMailbox = "pqr@stu.ltd.uk")
+
+        repository.save(ldu)
+        TestTransaction.flagForCommit()
+        TestTransaction.end()
+
+        TestTransaction.start()
+        val optionalOfLDU = repository.findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode)
+
+        assertThat(optionalOfLDU).isPresent
+        val persistentLdu = optionalOfLDU.get()
+        assertThat(persistentLdu.id).isNotNull()
+
+
+        // Business key equality
+        assertThat(persistentLdu).isEqualTo(ldu)
+
+        assertThat(ldu.createUserId).isEqualTo("anonymous")
+
+        // Check the db...
+        val count = lduCount(persistentLdu.id)
+        assertThat(count).isEqualTo(1)
+    }
+
+    @Test
+    fun `Persist Local Delivery Unit with Probation Teams`() {
+        val lduCode = uniqueLduCode()
+        val ldu: LocalDeliveryUnit2 = lduWithProbationTeams(lduCode)
+
+        repository.save(ldu)
+        TestTransaction.flagForCommit()
+        TestTransaction.end()
+
+        TestTransaction.start()
+        val optionalOfLdu = repository.findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode)
+        assertThat(optionalOfLdu).isPresent
+        val persistentLdu = optionalOfLdu.get()
+        assertThat(persistentLdu.id).isNotNull()
+
+        assertThat(persistentLdu.probationTeams).isEqualTo(lduWithProbationTeams(lduCode).probationTeams)
+
+        val id = persistentLdu.id
+        if (id != null) {
+            assertThat(probationTeamCount(id)).isEqualTo(2)
+        } else {
+            fail("No ID for persisted LDU")
+        }
```

Comment:
rather than having this style of logic in tests could use standard assertions: `assertThat(persistentLdu.id).isNotNull()`

Reply:
Yes